crates/fluvio-cli/src/connector/mod.rs

```diff
@@ -152,6 +155,41 @@ pub struct ProducerParameters {
     batch_size: Option<ByteSize>,
 }
 
+#[derive(Debug, Clone, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "kebab-case")]
+enum SmartModuleType {
+    Filter,
+    Map,
+    Arraymap,
+    Aggregate,
+    FilterMap,
+}
+
+impl ToString for SmartModuleType {
+    fn to_string(&self) -> String {
+        match self {
+            SmartModuleType::Filter => "filter",
+            SmartModuleType::Map => "map",
+            SmartModuleType::Arraymap => "arraymap",
```

Comment:
`arraymap` or `array-map`?

Reply:
It looks like in `fluvio-connector` we are using arraymap: See https://github.com/infinyon/fluvio-connectors/blob/4b894d7bacd3c761712da681bab9c63dd847144a/rust-connectors/common/src/opt.rs#L83

Reply:
Sounds good.